zap/src/irgen/mod.rs

```diff
@@ -1,11 +1,29 @@
 #![allow(dead_code)]
+use std::collections::HashMap;
 use std::{fmt::Display, vec};
 
 use crate::config::{NumTy, Range, Ty};
 
 pub mod des;
 pub mod ser;
 
+pub trait VarOccurrencesProvider {
```

Comment:
Why did you choose to use a seperate trait instead of making it a part of `Gen`?